sdk/core/azure-core/test/ut/websockets.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <gtest/gtest.h>
+
+#include <azure/core/internal/io/null_body_stream.hpp>
+#include <azure/core/websockets/websockets.hpp>
+
+#include <string>
+#include <utility>
+#include <vector>
+
+using namespace Azure::Core;
+using namespace Azure::Core::WebSockets;
+using namespace Azure::Core::IO;
+
+namespace Azure { namespace Core { namespace Test {
+
+  TEST(WebSocket, basicTest)
+  {
+    // websocket client creation
+    WebSocketClient wsClient(Url("ws://someUrl"));
+    wsClient.Connect();
+
+    // Callback for on received message
+    auto const onMessageReceived = [](WebSocketInMessage const& message) -> void { (void)message; };
+    wsClient.OnMessage(onMessageReceived);
+
+    // Sending a message
+    Azure::Core::IO::_internal::NullBodyStream noMessage;
+    WebSocketOutMessage outMessage(WebSocketMessageType::Ping, noMessage);
+    wsClient.Send(outMessage, Context{});
+
```

Comment:
As we mentioned earlier, this design drives a significant amount of complexity into WebSocket clients. Specifically, they need to understand how to implement concurrent programming, and we run afoul of all the concurrency issues associated with it. 

I'm quite concerned about the `onMessageReceived` lambda, since it's distressingly easy to abuse lambdas in multithreaded programs. It might make more sense to have the customer implement a class and have the OnMessage method take an instance of that class.